hack/util.sh

```diff
@@ -108,6 +108,4 @@ capz::util::generate_ssh_key() {
 capz::util::ensure_azure_envs() {
     : "${AZURE_SUBSCRIPTION_ID:?Environment variable empty or not defined.}"
     : "${AZURE_TENANT_ID:?Environment variable empty or not defined.}"
-    : "${AZURE_CLIENT_ID:?Environment variable empty or not defined.}"
```

Comment:
Isn't client id still required with workload identity? I see it's being kept in other places such as https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/4765/files#diff-c2ee8653e1d6b85f0aadf87cd438a9250806c052877248442be4d434cbc52425R37

Reply:
It's not required, no